tutorials/cassandra/src/main/java/com/datatorrent/cassandra/CassandraDataPopulator.java

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright (c) 2016 DataTorrent, Inc.
+ * All rights reserved.
+ */
+package com.datatorrent.cassandra;
+
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.ConsistencyLevel;
+import com.datastax.driver.core.PreparedStatement;
+import com.datastax.driver.core.Session;
+import com.datastax.driver.core.Statement;
+import com.datastax.driver.core.exceptions.DriverException;
+import com.datatorrent.api.Context;
+import com.datatorrent.api.Context.OperatorContext;
+import com.datatorrent.api.DefaultInputPort;
+import com.datatorrent.api.DefaultOutputPort;
+import com.datatorrent.api.annotation.InputPortFieldAnnotation;
+import com.datatorrent.api.annotation.OutputPortFieldAnnotation;
+import com.datatorrent.contrib.cassandra.CassandraPOJOOutputOperator;
+import com.datatorrent.contrib.cassandra.CassandraTransactionalStore;
+
+/**
+ * Extends {@link CassandraPOJOOutputOperator} to write tuples to cassandra database.<br/>
+ * This operator keeps track of hashcode of all written data tuples and emits
+ * ScanInformation which can be used for data validation.<br/>
+ * Operator Ports:<br/>
+ * 1. tuplesInput: Receives pojo tuples<br/>
+ * 2. ValidationInfo: emits {@link ScanInformation} required for data validation
+ * by downstream operators
+ *
+ */
+public class CassandraDataPopulator extends CassandraPOJOOutputOperator
+{
+  private int numRowsToWrite;
+  private int processedTuples;
+  private int hashCode;
+  private boolean writeMoreRecords = true;
+
+  @OutputPortFieldAnnotation(optional = true)
+  public transient DefaultOutputPort<ScanInformation> validatationInfo = new DefaultOutputPort<>();
+
+  @InputPortFieldAnnotation(optional = true)
+  public final transient DefaultInputPort<TestUsers> tuplesInput = new DefaultInputPort<TestUsers>()
```

Comment:
I assume this is needed because the process() method of the input port could not be overridden. Can you create a JIRA to abstract out the process method into processTuple() so that this can be called by the extending classes and re-declaration of ports is not needed.